include/RevMem.h

```diff
@@ -94,7 +94,7 @@ class RevMem {
     bool contains( const uint64_t& vBaseAddr, const uint64_t& Size ) {
       // exclusive top address
       uint64_t vTopAddr = vBaseAddr + Size - 1;
-      return ( this->contains( vBaseAddr ) && this->contains( vTopAddr ) );
+      return contains( vBaseAddr ) && contains( vTopAddr );
```

Comment:
Is the version with `this` incorrect? Or is the new version more correct

Reply:
Both are correct and the same, but the `this->` is redundant and makes it harder to read. `this->` is implicit  when accessing members of a class or its base classes from within a class's own member functions.

About the only time you need to use `this->` is when you have a `template` class trying to access a base class's members when the base class is dependent on its template parameter(s). Or if you have a local variable shadowing a class variable/function and you want to access the class variable/function instead of the local variable.

Reply:
(I know this PR is closed, but just some education:)

```c++
template<typename T>
class base {
public:
  bool mymember(){
    return true;
  }
};

template<typename T>
class derived : public base<T> {
public:
  bool something(){
    return mymember();         // Does not work: base class is dependent on template parameter T and won't be found
    return this->mymember();   // Works. "this" establishes the template parameter dependency
  }
};

```

Reply:
Makes a lot of sense!